OpenSim/Common/Exception.h

```diff
@@ -67,14 +110,31 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
```

Comment:
By non-invasive, I mean the PR should not affect behavior of any existing code. This PR is fundamentally about adding DataTable to the code base without having any code use it just yet. 

I will be making changes to other parts of code in subsequent PRs as I work on integrating DataTable into the code base. I also have another PR about DataAdapter almost ready for review but needs DataTable to be merged in before that. More work on this PR will just delay everything in the pipeline.

In my understanding changing Exception class's access specifiers is not part of DataTable PR. I think this PR has already been way too long in the making. I am hoping to merge it without any more stretching.

I will be opening as many PRs as required to implement improvements, including ones to improve exceptions.